src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/StringCollectionTextWriter.cs

```diff
@@ -62,7 +70,19 @@ public override void Write([NotNull] char[] buffer, int index, int count)
                 throw new ArgumentOutOfRangeException(nameof(count));
             }
 
-            Content.Append(buffer, index, count);
+            while (count > 0)
+            {
+                // Split large char arrays into 1KB strings.
+                var currentCount = count;
+                if (MaxCharToStringLength < currentCount)
+                {
+                    currentCount = MaxCharToStringLength;
+                }
+
+                _content.Append(new string(buffer, index, currentCount));
+                index += currentCount;
+                count -= currentCount;
+            }
```

Comment:
This method is part of the textwriter API, which is why we implement it here. The better change would be to remove it from `BufferedHtmlContent`.

Thoughts?